pxr/usd/sdf/listProxy.h

```diff
@@ -136,43 +135,129 @@ class SdfListProxy {
     friend class _ConstGetHelper;
 
     template <class Owner, class GetItem>
-    class _Iterator :
-        public boost::iterator_facade<
-            _Iterator<Owner, GetItem>,
-            std::remove_cv_t<
+    class _Iterator {
+        class _PtrProxy {
+        public:
+            std::add_pointer_t<typename GetItem::result_type> operator->() {
+                return std::addressof(_result);
+            }
+        private:
+            friend class _Iterator;
+            explicit _PtrProxy(
+                std::add_const_t<
+                    std::add_lvalue_reference_t<
+                        typename GetItem::result_type>
+                > result) : _result(result) {}
+            typename GetItem::result_type _result;
+        };
+    public:
+        using This = _Iterator<Owner, GetItem>;
+        using iterator_category = std::random_access_iterator_tag;
+        using value_type = std::remove_cv_t<
                 std::remove_reference_t<
                     typename GetItem::result_type
                 >
-            >,
-            std::random_access_iterator_tag,
-            typename GetItem::result_type> {
-    public:
-        typedef _Iterator<Owner, GetItem> This;
-        typedef
-            boost::iterator_facade<
-                _Iterator<Owner, GetItem>,
-                std::remove_cv_t<
-                    std::remove_reference_t<
-                        typename GetItem::result_type
-                    >
-                >,
-                std::random_access_iterator_tag,
-                typename GetItem::result_type> Parent;
-        typedef typename Parent::reference reference;
-        typedef typename Parent::difference_type difference_type;
-
-        _Iterator() : _owner(NULL), _index(0)
-        {
-            // Do nothing
-        }
+            >;
+        using reference = typename GetItem::result_type;
+        using pointer = _PtrProxy;
+        using difference_type = std::ptrdiff_t;
+
+        static_assert(!std::is_lvalue_reference<reference>::value,
+                      "_PtrProxy is only needed when reference type alias is "
+                      "not be a lvalue reference type");
+
+        _Iterator() = default;
 
         _Iterator(Owner owner, size_t index) : _owner(owner), _index(index)
         {
             // Do nothing
         }
 
+        reference operator*() const { return dereference(); }
```

Comment:
I'm confused as to why this -> returns the address directly when you define pointer = _PtrProxy and static_assert that value type of reference is not an lvalue. My expectation is that your assert would mean we always return the _PtrProxy or that we want a conditional type here like you did SdfMapEditProxy

Reply:
You're right!  That was an error.  Thanks.